app/controllers/admin/enterprise_fees_controller.rb

```diff
@@ -78,5 +78,15 @@ def redirect_path
 
       main_app.admin_enterprise_fees_path
     end
+
+    def enterprise_fee_bulk_params
+      params.require(:enterprise_fee_set).permit(
+        collection_attributes: [
+          :id, :enterprise_id, :fee_type, :name, :tax_category_id,
+          :inherits_tax_category, :calculator_type,
+          { calculator_attributes: [:id, :preferred_flat_percent] }
+        ]
+      )
+    end
```

Comment:
I dont really understand why this is needed here and specially why it is not seen in rails 4.0 This bulk update action is definitely working in master. My best guess is that it's because of the messy manipulation that is done in ModelSet and how collection_attributes= is called?

Reply:
Everything related to collections in params was totally broken in Rails 4.2 with lots of `ForbiddenAtributes` errors. Interestingly, when I tried to pull these changes into master, it also broke the specs there. I think maybe some logic has changed around collections and strong params?

> the messy manipulation that is done in ModelSet

Yeah, there's a few places where we have some really nasty code around params manipulation. I was going for a quick "fix the issues causing broken specs and move on" approach at the time, but we could do some refactoring.

Reply:
ok :+1: 
Now I wonder if this was not just caused by moving api controllers from metal to base. Maybe Strong Parameters are enforced only in Base, not Metal.

Reply:
Yes, that's it. Metal is missing lots of things, and Strong Params is one of them, but we didn't realise before because there were no errors :+1: